### PR TITLE
Added Jito-RPC Executer

### PR DESCRIPTION
**Pull request by mcoder9**
Added Jito-RPC Executer
https://jito-labs.gitbook.io/mev/searcher-resources/json-rpc-api-reference

No need to authorize custom Private Key from jito
Just set the executer to 'jito' and set the fees

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -80,8 +81,9 @@ function printDetails(wallet: Keypair, quoteToken: Token, bot: Bot) {
   logger.info('- Bot -');
 
   logger.info(`Using warp: ${bot.isWarp}`);
+  logger.info(`Using jito: ${bot.isJito}`);
   if (bot.isWarp) {
```
Comment (fdundjer) [inline index.ts:85]: Should check here if it's jito

File: transactions/jito-rpc-transaction-executor.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        logger.debug(`Confirming jito transaction...`);
+        return await this.confirm(jitoTxsignature, latestBlockhash);
+      } else {
+        logger.debug(`No successful responses received for jito`);
+      }
+
+      return { confirmed: false };
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        logger.trace({ error: error.response?.data }, 'Failed to execute warp transaction');
```
Comment (fdundjer) [inline transactions/jito-rpc-transaction-executor.ts:112]: should be: 'Failed to execute jito transaction'

File: index.ts (hunk 1 above)
Comment (mcoder9) [reply index.ts:86]: fixed

File: transactions/jito-rpc-transaction-executor.ts (hunk 2 above)
Comment (mcoder9) [reply transactions/jito-rpc-transaction-executor.ts]: fixed

**Review by fdundjer — APPROVED** [review]

**Comment by thedotmack** [comment]
What if we want to use a custom RPC with our own jito setup? I tried today and couldn’t get it to work

**Comment by mcoder9** [comment]
This setup for jito doesn't support custom RPC.
If it doesn't work it is because of congestion or low tips/slippage.
The code it self is well implemented.

**Comment by thedotmack** [comment]
I hear you, I think it should support custom RPC with the jito keypair because it would be faster, no?

**Comment by thedotmack** [comment]
> The code it self is well implemented.

This feels like a challenge 😳

**Pull request merged**